pkg/npm/package_manager.go

```diff
@@ -66,8 +60,28 @@ func (exec *Execute) detectPackageManager() (*PackageManager, error) {
 		}
 		if exists {
 			if pm.Name == "pnpm" {
-				if err := pm.InstallPnpm(exec.Utils.GetExecRunner()); err != nil {
-					return nil, fmt.Errorf("failed to install pnpm: %w", err)
+				execRunner := exec.Utils.GetExecRunner()
+
+				// Check if pnpm is globally installed first
+				if err := execRunner.RunExecutable("pnpm", "--version"); err == nil {
+					// Use global pnpm
+					pm.InstallCommand = "pnpm"
+					log.Entry().Info("Using globally installed pnpm")
+				} else {
+					// Check if pnpm is locally installed
+					if err := execRunner.RunExecutable(pnpmPath, "--version"); err == nil {
+						// Use local pnpm
+						pm.InstallCommand = pnpmPath
+						log.Entry().Info("Using locally installed pnpm")
```

Comment:
What if locally/globally installed version is different than pnpmVersion mentioned by user?